src/app/App.tsx

```diff
@@ -1,23 +1,53 @@
 import React, { useState } from 'react';
+import Card from './components/Card/Card';
 import Form from './components/Form/Form';
 
 type Pupil = {
   name: string;
-  evaluation: string;
+  evaluations: string[];
 };
 
 function App(): JSX.Element {
   const [pupils, setPupils] = useState<Pupil[]>([]);
   console.log(pupils);
 
+  function findPupil(pupil: { name: string; evaluation: string }) {
+    const newPupil = pupil;
+    const existingPupil = pupils.find((pupil) => pupil.name === newPupil.name);
+    if (existingPupil === undefined) {
+      return false;
+    } else {
+      return existingPupil;
+    }
+  }
```

Comment:
```suggestion
  function findPupilByName(name: string) {
    return pupils.find((pupil) => pupil.name === name);
  }
```

Reply:
Great, I knew it was way too complicated :D